### PR TITLE
Customization (DT) Beta - Font Icon Picker

### PR DESCRIPTION
**Pull request by kodinkat**
Screenshot 2023-05-23 at 16 25 40

### CODE REVIEW

**Comment by prykon** [comment]
Tile Preview doesn't update on new field creation, but this issue appears to exist from before this branch.

Aside from my previous comment, no errors were found. Functionality works.

Nice job, @kodinkat !

**Review by prykon — CHANGES_REQUESTED** [review]
@kodinkat Thanks so much for coding this! I'm testing it as we speak.

In order to keep the UI consistent, could you...?

- [ ] Make the edit field settings modal card flip and make the icon selector modal unflip
- [ ] Add the `dt-admin-modal-overlay` class when opening the icons modal

Thanks again!

**Comment by kodinkat** [comment]
> @kodinkat Thanks so much for coding this! I'm testing it as we speak.
> 
> In order to keep the UI consistent, could you...?
> 
> * [ ]  Make the edit field settings modal card flip and make the icon selector modal unflip
> * [ ]  Add the `dt-admin-modal-overlay` class when opening the icons modal
> 
> Thanks again!

@prykon That's now in for you... Updating font icon picker dialog to adopt flip animation turned out to be slightly fiddlier then anticipated; so let me know if anything...

**Review by prykon — APPROVED** [review]

**Comment by prykon** [comment]
@kodinkat I would add a `margin: 1.5em` to the `.ui-dialog-buttonset`
Aside from that, everything looks good!

**Comment by prykon** [comment]
I think we're good to merge, @corsacca

**Comment by kodinkat** [comment]
@prykon @corsacca - See if recent updates now address multi-select mdi icon preview issues.

**Comment by prykon** [comment]
@kodinkat Icon updates properly on the admin page and dt frontend but not if it's inside a multi select...

**This is ok**
Screenshot 2023-06-05 at 10 40 39

**This is broken**
Screenshot 2023-06-05 at 10 42 04

**Comment by kodinkat** [comment]
@prykon try it now and let me know.. :)

**Comment by corsacca** [comment]
@prykon, That looks like an issue with the Church Health and not with the implementation of the font icon picker. Maybe a separate issue?

File: dt-core/global-functions.php [hunk 1]
```diff
@@ -325,11 +325,18 @@ function dt_render_field_icon( $field, $class = 'dt-icon', $default_to_name = fa
                 } else {
                     $alt_tag = '';
                 }
-                ?>
-
-                <img class="<?php echo esc_html( $class ); ?>" src="<?php echo esc_url( $field['icon'] ) ?>" alt="<?php echo esc_html( $alt_tag ) ?>" width="15" height="15">
 
-                <?php
+                // Determine icon html element shape to be adopted.
```
Comment (corsacca) [inline dt-core/global-functions.php:329]: @kodinkat, font icons need to use the `font-icon` field, not the `icon` field, then the code at line 340 works

**Comment by kodinkat** [comment]
@prykon @corsacca give it a try now and let me know how things hold up, especially on the group church health side of things! :)

**Comment by prykon** [comment]
@kodinkat Looks good now!

**Comment by kodinkat** [comment]
> @kodinkat @prykon looking good guys. @kodinkat i tried adding a new field and got: image There is a trim() error, but then show:
> 
> this is what a new field (multi_select) looks like: image

@corsacca sanity checked new field / option flows and both held true for me. @prykon does it hold true for you?

Screenshot 2023-06-07 at 10 20 05

Screenshot 2023-06-07 at 10 20 56

Screenshot 2023-06-07 at 10 24 09

Screenshot 2023-06-07 at 10 25 38

Screenshot 2023-06-07 at 10 27 17

**Comment by prykon** [comment]
@kodinkat Thanks for hanging in there for this one; so many ins and outs!

I found a bug...
Screenshot 2023-06-07 at 09 50 45

**Steps to reproduce:**

1. Create a new field
2. Click to edit the field (no icon image is shown, but this might be out of scope)
3. Edit the field name only (don't change icon settings)
4. Re-click to edit the field (icon image now appears but is broken).

**Comment by kodinkat** [comment]
> @kodinkat Thanks for hanging in there for this one; so many ins and outs!
> 
> I found a bug... Screenshot 2023-06-07 at 09 50 45
> 
> **Steps to reproduce:**
> 
> 1. Create a new field
> 2. Click to edit the field (no icon image is shown, but this might be out of scope)
> 3. Edit the field name only (don't change icon settings)
> 4. Re-click to edit the field (icon image now appears but is broken).
> 
> Apparently, it's saving the icon value to _undefined_ and that also breaks the frontend... Screenshot 2023-06-07 at 09 59 40

@prykon thanks for this and for the steps; which helped me to reproduce! A fix has now been pushed, so feel free to have a play and let me know if there are still any remaining hidden bugs..... :)

**Comment by prykon** [comment]
@kodinkat
Found a UI thing... the 'Icon' button should not be on a new line.
Screenshot 2023-06-08 at 10 24 12

@corsacca Do we want icons on field options for a dropdown field type?

Everything else looks great!

**Comment by prykon** [comment]
Looking good, @kodinkat!
I think it's requiring an onclick prevent default or something.

https://github.com/DiscipleTools/disciple-tools-theme/assets/36511666/c8600591-0ca8-4ff5-8e49-a1bd2419dc04

**Comment by kodinkat** [comment]
> Looking good, @kodinkat! I think it's requiring an onclick prevent default or something.
> 
>  Screen.Recording.2023-06-09.at.09.17.31.mov

@prykon thanks for this... However, issue not showing up on my end for `.change-icon-button` button class types or any other buttons..

Anyway, just to be on the safe-side, I've added `e.preventDefault()` on `.change-icon-button` click events; but not on any others, to avoid disrupting any pre-existing logic..

**Comment by prykon** [comment]
> > Looking good, @kodinkat! I think it's requiring an onclick prevent default or something.
> > Screen.Recording.2023-06-09.at.09.17.31.mov
> 
> @prykon thanks for this... However, issue not showing up on my end for `.change-icon-button` button class types or any other buttons..
> 
> Anyway, just to be on the safe-side, I've added `e.preventDefault()` on `.change-icon-button` click events; but not on any others, to avoid disrupting any pre-existing logic..

@kodinkat  It's working now! Also, no other bugs found.

@corsacca are we good to push?

**Pull request merged**